spring-kafka-test/src/main/java/org/springframework/kafka/test/EmbeddedKafkaKraftBroker.java

```diff
@@ -0,0 +1,593 @@
+	 * @param partitions partitions per topic.
+	 * @param topics the topics to create.
+	 */
+	public EmbeddedKafkaKraftBroker(int count, int partitions, String... topics) {
+		this.count = count;
+		this.kafkaPorts = new int[this.count]; // random ports by default.
+		if (topics != null) {
+			this.topics = new HashSet<>(Arrays.asList(topics));
+		}
+		else {
+			this.topics = new HashSet<>();
+		}
+		this.partitionsPerTopic = partitions;
+	}
+
+	/**
+	 * Specify the properties to configure Kafka Broker before start, e.g.
+	 * {@code auto.create.topics.enable}, {@code transaction.state.log.replication.factor} etc.
+	 * @param properties the properties to use for configuring Kafka Broker(s).
+	 * @return this for chaining configuration.
+	 * @see KafkaConfig
+	 */
+	@Override
+	public EmbeddedKafkaBroker brokerProperties(Map<String, String> properties) {
+		this.brokerProperties.putAll(properties);
+		return this;
+	}
+
+	/**
+	 * Specify a broker property.
+	 * @param property the property name.
+	 * @param value the value.
+	 * @return the {@link EmbeddedKafkaKraftBroker}.
+	 */
+	public EmbeddedKafkaBroker brokerProperty(String property, Object value) {
+		this.brokerProperties.put(property, value);
+		return this;
+	}
+
+	/**
+	 * Set explicit ports on which the kafka brokers will listen. Useful when running an
+	 * embedded broker that you want to access from other processes.
+	 * @param ports the ports.
+	 * @return the {@link EmbeddedKafkaKraftBroker}.
+	 */
+	@Override
+	public EmbeddedKafkaKraftBroker kafkaPorts(int... ports) {
+		Assert.isTrue(ports.length == this.count, "A port must be provided for each instance ["
+				+ this.count + "], provided: " + Arrays.toString(ports) + ", use 0 for a random port");
+		this.kafkaPorts = Arrays.copyOf(ports, ports.length);
+		return this;
+	}
+
+	/**
+	 * Set the system property with this name to the list of broker addresses.
+	 * @param brokerListProperty the brokerListProperty to set
+	 * @return this broker.
+	 * @since 2.3
+	 */
+	@Override
+	public EmbeddedKafkaBroker brokerListProperty(String brokerListProperty) {
+		this.brokerListProperty = brokerListProperty;
+		return this;
+	}
+
+	/**
+	 * Set the timeout in seconds for admin operations (e.g. topic creation, close).
+	 * @param adminTimeout the timeout.
+	 * @return the {@link EmbeddedKafkaKraftBroker}
+	 * @since 2.8.5
+	 */
+	public EmbeddedKafkaBroker adminTimeout(int adminTimeout) {
+		this.adminTimeout = Duration.ofSeconds(adminTimeout);
+		return this;
+	}
+
+	/**
+	 * Set the timeout in seconds for admin operations (e.g. topic creation, close).
+	 * Default 10 seconds.
+	 * @param adminTimeout the timeout.
+	 * @since 2.2
+	 */
+	public void setAdminTimeout(int adminTimeout) {
+		this.adminTimeout = Duration.ofSeconds(adminTimeout);
+	}
+
+	@Override
+	public void afterPropertiesSet() {
+		overrideExitMethods();
+		addDefaultBrokerPropsIfAbsent(this.brokerProperties, this.count);
+		start();
```

Comment:
If you start it from the `afterPropertiesSet()` why do we need a `SmartLifecycle` contract then?
Why cannot we stop everything in the `destroy()` as it is for ZK variant?